airflow/dag_processing/manager.py

```diff
@@ -711,7 +683,10 @@ def clear_nonexistent_import_errors(self, session=NEW_SESSION):
         query = delete(ParseImportError).where(ParseImportError.processor_subdir == self.get_dag_directory())
 
         if self._file_paths:
-            query = query.where(ParseImportError.filename.notin_(self._file_paths))
+            query = query.where(
+                ~ParseImportError.filename.in_(self._file_paths),
```

Comment:
nit:

`notin_` feels more explicit imo.. but I think I know why you did that, since we now have 2 conditions, easier to identify what isn't vs what is reqd

```suggestion
                ParseImportError.filename.notin_(self._file_paths),
```

Reply:
Yeah agreed - I changed some others (or maybe it was on a changed it back).